### PR TITLE
Status Chart: chart.js 3.6.0, improve type info, weekly updates

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* npm update.
* README.md: Update Node.js version.
* chart.js 3.6.0.
  + This fixes chartjs/Chart.js#9678, noted in #2100.
* weekly_table.js updates.
* gather_stats.ts: Improve type info.
  + Various raw fields can be `null`. (For `author`, this is when a user has been deleted; for `closedAt`/`mergedAt`, this is when a PR/issue hasn't been closed/merged.) We're already testing for `null`, this just acknowledges that in the type system.
  + Add types for the `RateLimit` and combined `RawNodes` that `retrieve_nodes_from_network()` and `retrieve_nodes()` return.
  + As we gradually build up the information for `RateLimit`, TypeScript is intelligent enough to see that `spent`/`remaining`/`limit` will always be `number` instead of `null` by the time we return, making the `RateLimit` type more useful. To achieve this, we need individual variables in `retrieve_nodes_from_network()`. We can say that `spent`/`remaining`/`limit` are `number` without an initializer, and TypeScript will require that we assign to them before using them. We need to change the `while` loop into a `do/while` loop so that TypeScript will see that it always runs at least one iteration (this was already guaranteed because `variables.want_prs` and `variables.want_issues` are initially `true`).
  + In `retrieve_nodes()`, we need a type assertion (`as RawNodes`) to tell the compiler what we're loading.
  + In `warn_about_duplicates()`, constructing `Set<string>` tells TypeScript what we're doing (otherwise we're working with `Set<any>`).
  + Similarly, in `write_monthly_table()`, we're really using a `Map<string, number>`.
  + After thinking about it, I've changed `transform_pr_nodes()` and `transform_issue_nodes()` to have deduced return types (with explicitly typed `ret` variables). This is more consistent with the rest of the code now, and while both ways are equally type-checked, the `ret` variables offer better diagnostics if the types are wrong (as TypeScript doesn't have to complain about the huge transformation in the function).

:chart_with_upwards_trend: Live preview: [stephantlavavej.github.io/STL/](https://stephantlavavej.github.io/STL/)
===

### CODE REVIEW

File: src/gather_stats.ts [hunk 1]
```diff
@@ -106,11 +120,8 @@ async function retrieve_nodes_from_network() {
...
 
-        const rate_limit = {
-            spent: null as null | number,
-            remaining: null as null | number,
-            limit: null as null | number,
-        };
+        let spent: number;
+        let limit: number;
 
         {
```
Comment (AlexGuteniev) [inline src/gather_stats.ts:126]: (Pre-existing) I know Javascript does not have scope by default, so variable declared in scope can be accessed from anywhere in a function after or before the scope. I hope TypeScript respects scope.

File: src/gather_stats.ts [hunk 2]
```diff
@@ -106,11 +120,8 @@ async function retrieve_nodes_from_network() {
             headers: { authorization: `token ${process.env.SECRET_GITHUB_PERSONAL_ACCESS_TOKEN}` },
         });
 
-        const rate_limit = {
-            spent: null as null | number,
-            remaining: null as null | number,
-            limit: null as null | number,
```
Comment (AlexGuteniev) [inline src/gather_stats.ts:112]: `null as null` is funny.

**Review by AlexGuteniev — APPROVED** [review]

File: src/gather_stats.ts (hunk 1 above)
Comment (AlexGuteniev) [reply src/gather_stats.ts:126]: Aha, I see, with `let` or `const` the block scope is respected!

Comment (StephanTLavavej) [reply src/gather_stats.ts:126]: Yep! The Status Chart has always avoided `var` for this reason.

**Review by CaseyCarter — APPROVED** [review]

File: src/gather_stats.ts (hunk 2 above)
Comment (StephanTLavavej) [reply src/gather_stats.ts:112]: Yeah, it looks a little strange :joy_cat: The type is `null | number`, so this should be read with `as` having lower/weaker precedence. This pattern still occurs in `null as null | cliProgress.SingleBar` - the order is intended to indicate "this is `null` now, but will be a specific type later".

Elsewhere, I wrote `closedAt: string | null`. There, the order indicates "this is usually `string` but can sometimes be `null`".

**Pull request merged**